app/src/main/java/waiter/Communicator.java

```diff
@@ -0,0 +1,11 @@
+package waiter;
+
+import java.io.IOException;
+
+public class Communicator {
+    void communicate(Awaiter server, Transporter messenger) throws IOException {
+        Connection client = server.awaitClient();
```

Comment:
Same with my other comment, let's call this `connection` instead of of `client` since somewhere in the code we're using `Socket client`.

Reply:
I was initially avoiding naming the variables after their types because it seemed redundant, I discovered the `var` keyword and think an ideal solution would be to use this to name them after what they are, what are your thoughts?

Reply:
Can you clarify what you meant by using the var keyword?

Reply:
Like using `var connection` instead of `Connection connection`, the var connection still is of type Connection in this case, just less redundant reading wise

Reply:
Either way is really ok to me, just thought it might look a little more clean, but maybe it then is less understandable from a reading perspective, what are your thoughts?

Reply:
I see. I think I'm leaning toward Connection connection. It's quite common that the variable name is the same as the type in Java so you should be ok 😄

Reply:
+1 to @mrfishball I would follow the convention of `Connection connection`. I don't have a strong preference for/against using `var`/leaning on Local-Variable Type Inference unless it makes the code less readable.

Reply:
Resolved in [this commit](https://github.com/jonmanahan/waiter/pull/2/commits/c5b65f8f2b576acc81c13d122850ed83f4052bda), if it resolves the issues please hit resolve conversation if possible, thank you all a ton!